client/src/client.ts

```diff
@@ -2561,6 +2568,25 @@ export abstract class BaseLanguageClient {
 		this._name = name;
 
 		clientOptions = clientOptions || {};
+
+		let resolvedCancellationFolderName: string | undefined;
+		const uniqueName = getUniqueName(clientOptions.cancellationFolderName);
```

Comment:
I have put creation and removal of the cancellation folder in language client (basically client owns that folder), but wondering whether I should put it in client connection.. or might not matter.

Reply:
sure. let me change it to boolean and do what you said.